aiohomekit/controller/ble/pairing.py

```diff
@@ -865,6 +865,38 @@ async def _close_while_locked(self) -> None:
     async def list_accessories_and_characteristics(self) -> list[dict[str, Any]]:
         return self.accessories.serialize()
 
+    async def get_primary_name(self) -> str:
+        """Return the primary name of the device.
+
+        This overrides the default implementation
+        to get name from the advertisement if the
+        accessories are not available which happens
+        frequently with the ESPHome proxies because
+        they cannot currently connect right away again
+        after disconnecting.
+
+        We want to avoid raising here if possible
+        since this is called right after pairing
+        and if we raise the pairing will fail but
+        the device will still be paired and the user
+        will have to un-pair it manually with a factory
+        reset.
+        """
+        if not self.accessories:
+            if self.description:
+                self._accessories_state = AccessoriesState(
+                    Accessories(), -1, self.broadcast_key
+                )
+                return self.description.name
```

Comment:
I think we could put this at the top and than pass to super() to reduce the duplicate code